### PR TITLE
fix(retype): optionals in extracted types

### PR DESCRIPTION
**Pull request by goldjacobe**
this one ended up being just shapes, did tagged unions upstack

### CODE REVIEW

**Comment by goldjacobe** [comment]
Current dependencies on/for this PR:
* main
  * **PR #6** <a href="https://app.graphite.dev/github/pr/withgraphite/retype/6" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/withgraphite/retype/6?utm_source=stack-comment).

**Review by tomasreimers — APPROVED** [review]

**Comment by goldjacobe** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/withgraphite/retype/6?utm_source=stack-comment).


Job Reference: pXLNmN6X5NmFIiJrO8xZ

**Pull request merged**
